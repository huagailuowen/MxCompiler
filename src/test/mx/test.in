
=============================== Build time: 6ms ================================

49

============================= Interpret time: 33ms =============================

0
