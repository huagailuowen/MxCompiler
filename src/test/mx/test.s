  .data
  .rodata
@string.2:
  .string "\nabbdbdd\\\\\"d\"c\\adc\"c\00"
@string.0:
  .string "\"\nbda\\\"\ndbd\\c\"\\\"c\\c\"\00"
@string.1:
  .string "\\ab\nac\"\"add\n\"\nac\ndb\"\00"
  .text
  .globl __init__
__init__:
  addi   sp, sp, -112
  sw    ra, 0(sp)

__init___recursive:
  lw    ra, 0(sp)
  addi   sp, sp, 112
  ret   


  .globl main
main:
defaultStart.__init__._0:
  addi   sp, sp, -112
  sw    ra, 0(sp)
  sw    s11, 88(sp)

main_recursive:
  la    a0, @string.2
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  call  print
  la    a0, @string.2
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  call  print
  la    a0, @string.1
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  call  print
  la    a0, @string.1
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  call  print
  la    a0, @string.1
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  call  print
  la    a0, @string.0
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  call  print
  la    a0, @string.1
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  call  print
  la    a0, @string.0
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  call  print
  la    a0, @string.2
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.0
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  call  print
  la    a0, @string.0
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.1
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  la    a1, @string.2
  call  __string_concat
  mv    s11, a0
  mv    a0, s11
  call  print

func.main.end:
  li    a0, 0
  lw    ra, 0(sp)
  lw    s11, 88(sp)
  addi   sp, sp, 112
  ret   



