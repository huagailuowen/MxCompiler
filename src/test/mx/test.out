ptr %this.0.in : 3
i32 %loopvar.4.0._2 : 2
i32 %load.10 : 0
i32 %i.0._10 : 0
ptr %load.44 : -1
i32 %load.27 : -1
ptr %getele.7 : 2
ptr %getele.6 : 2
i32 %call.4 : 1
i32 %call.0 : 0
i32 %load.39 : 0
ptr %load.9 : -1
i32 %arith.5 : 0
ptr %malloc.6.addr : 1
i32 %load.4 : 0
ptr %load.29 : 2
ptr %getele.10 : 2
i32 %arith.9 : 1
ptr %getele.13 : 2
i32 %arith.7 : 2
ptr %load.26 : 2
i32 %load.37 : -1
i32 %arith.10 : 0
i32 %arith.0 : 3
ptr %getele.5 : 2
ptr %getele.1 : 3
ptr %malloc.4 : 1
i32 %arith.3 : 2
ptr %load.50 : 2
i32 %load.19 : 1
ptr %load.24 : 2
i32 %load.1 : -1
i32 %call.2 : 1
i32 %loopvar.7.0._14 : 2
i32 %call.3 : 1
i32 %i.0._18 : 0
ptr %getele.3 : 3
ptr %malloc.1 : 1
ptr %load.22 : -1
ptr %malloc.8 : 1
ptr %load.15 : -1
ptr %load.36 : 2
ptr %load.20 : 2
ptr %getele.8 : 2
ptr %load.31 : 2
ptr %getele.4 : 2
i32 %arith.2 : 3
i1 %arith.4 : 1
i32 %load.32 : -1
ptr %load.34 : 2
ptr %getele.12 : 3
i32 %load.52 : -1
i32 %arith.1 : 2
i1 %arith.8 : 1
ptr %getele.9 : 2
i32 %arith.6 : 3
i32 %call.1 : 0
i32 %load.3 : -1
i32 %loopvar.5.0._6 : 2
i32 %load.48 : 1
ptr %string.3 : 1
ptr %load.119 : 6
i32 %load.120 : 6
i32 %lpos.0._22 : -1
i32 %arith.22 : 6
i1 %arith.11 : 0
ptr %load.123 : 7
i32 %r.0.in : 4
ptr %load.128 : 5
ptr %load.130 : -1
ptr %string.9 : 1
ptr %getele.19 : 6
i32 %rpos.0._5 : 5
ptr %load.135 : 2
ptr %string.2 : 1
ptr %load.107 : 6
i32 %load.124 : 7
ptr %getele.15 : 6
i32 %arith.13 : 0
ptr %load.109 : 2
i1 %arith.17 : 6
ptr %load.147 : 6
ptr %load.117 : 6
ptr %getele.16 : 2
i1 %arith.15 : 6
ptr %load.95 : 6
ptr %load.151 : -1
ptr %getele.26 : 6
ptr %string.6 : 2
ptr %load.125 : 5
i32 %arith.23 : 6
i32 %l.0.in : 3
ptr %getele.24 : 2
ptr %load.138 : -1
i1 %arith.25 : 6
ptr %load.121 : 7
ptr %getele.22 : 2
i32 %i.1._22 : -1
i32 %lpos.0._16 : 2
ptr %getele.18 : 6
i32 %arith.24 : 1
i32 %arith.20 : 7
ptr %string.7 : 1
ptr %load.110 : 6
i32 %arith.26 : 6
i32 %arith.12 : 0
ptr %load.112 : -1
ptr %getele.21 : 7
i1 %arith.19 : 6
ptr %load.136 : 5
i32 %arith.28 : 1
ptr %string.5 : 1
ptr %load.148 : 7
ptr %load.133 : 2
i32 %i.1._18 : 1
ptr %load.145 : 6
i32 %lpos.0._5 : 2
ptr %getele.14 : 5
ptr %load.100 : -1
i32 %mid.0._22 : -1
ptr %getele.20 : 7
i32 %i.1._5 : 1
ptr %getele.27 : 7
ptr %getele.17 : 6
i32 %arith.16 : 6
ptr %load.97 : 5
i32 %arith.27 : 8
i32 %arith.18 : 7
i1 %arith.21 : 6
ptr %load.127 : 2
i32 %rpos.0._16 : 5
ptr %getele.25 : 5
ptr %load.98 : 6
i32 %rpos.0._22 : -1
i32 %arith.14 : 0
ptr %getele.23 : 5
i32 %load.156 : 0